Demo docker error build

## Environment
* OS Version: Ubuntu 22.04
* o3de binary build 23.10.2
* ROS2 Humble

## Description
When running Option A: Only For Demo!

./docker/build.sh demo

Receive an error
```
 => ERROR [demo 3/9] RUN /opt/O3DE/23.10.2/scripts/o3de.sh register --project-path /home/maxxlife/andino_o3de                                                                   0.3s
------
 > [demo 3/9] RUN /opt/O3DE/23.10.2/scripts/o3de.sh register --project-path /home/maxxlife/andino_o3de:
0.222 1.10.1
0.272 [ERROR] o3de.register: andino_o3de is not known to be compatible with the following objects/APIs and requires the --force parameter to register:
0.272   The following dependency requirements could not be satisfied:
0.272 The project requires ROS2
0.272 <Gem RosRobotSample==1.0.0> requires ROS2
0.272 <Gem RGL==1.1.0> requires ROS2>=3.2.0
------
Dockerfile:113
--------------------
 111 |     
 112 |     # Register the andino_o3de project
 113 | >>> RUN /opt/O3DE/$O3DE_VERSION/scripts/o3de.sh register --project-path $ANDINO_O3DE_PATH
 114 |     
 115 |     WORKDIR /home/${USERNAME}/
--------------------
ERROR: failed to solve: process "/bin/sh -c /opt/O3DE/$O3DE_VERSION/scripts/o3de.sh register --project-path $ANDINO_O3DE_PATH" did not complete successfully: exit code: 1
```

--- a/docker/o3de.repos
+++ b/docker/o3de.repos
@@ -6,7 +6,7 @@ repositories:
   o3de-rgl-gem:
     type: git
     url: https://github.com/RobotecAI/o3de-rgl-gem.git
-    version: development
+    version: 23f92afca6d8842988969cc130313910e3bf7aa5
   RobotVacuumSample:
     type: git
     url: https://github.com/o3de/RobotVacuumSample.git

